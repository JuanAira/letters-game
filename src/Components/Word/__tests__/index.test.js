import React from 'react';
import { render } from '@testing-library/react';
import { Word } from '../index';
import SharedProps from '../../../context/SharedProps';

describe('<Word /> component', () => {
  const setup = ({ word }) => {
    const sharedProps = {
      word,
    };

    const component = render(
      <SharedProps.Provider value={sharedProps} >
        <Word />
      </SharedProps.Provider>
    );

    return component;
  };

  it('should match snapshot - word valid', () => {
    const { container } = setup({ word: 'TOY'});
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot - word invalid', () => {
    const { container } = setup({ word: 'DEMO' });
    expect(container).toMatchSnapshot();
  });
});
