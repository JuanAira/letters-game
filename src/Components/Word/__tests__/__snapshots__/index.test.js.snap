// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Word /> component should match snapshot - word invalid 1`] = `
<div>
  <h3
    class="word word--invalid"
  >
    <span
      class="word--formed"
    >
      DEMO
    </span>
    <span
      class="word--feedback"
    >
      invalid
    </span>
  </h3>
</div>
`;

exports[`<Word /> component should match snapshot - word valid 1`] = `
<div>
  <h3
    class="word word--valid"
  >
    <span
      class="word--formed"
    >
      TOY
    </span>
    <span
      class="word--feedback"
    >
      valid
    </span>
  </h3>
</div>
`;
